Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Themes/ThemeStateProperty.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    /// <summary>
+    /// Class to store information about a Theme property that contains values per available state
+    /// </summary>
+    [Serializable]
+    public class ThemeStateProperty
+    {
+        // TODO: Troy Add comments
+        public string Name;
+        public ThemePropertyTypes Type;
+        public List<ThemePropertyValue> Values;
+        public ThemePropertyValue StartValue;
+        public ThemePropertyValue Default;
+
+        public Shader TargetShader;
+        public string ShaderPropertyName;
+
+        [System.NonSerialized]
+        protected int ShaderPropertyID = -1;
+
+        private static readonly ThemePropertyTypes[] ShaderTypes =
+            { ThemePropertyTypes.Color, ThemePropertyTypes .ShaderFloat, ThemePropertyTypes.ShaderRange};
+
+        public static bool IsShaderPropertyType(ThemePropertyTypes type)
+        {
+            return ShaderTypes.Contains(type);
+        }
+
+        /// <summary>
+        /// Lazy loads shader property ID from Unity for the ShaderPropertyName
+        /// </summary>
+        /// <returns>integer key for current shader property to get/set shader values</returns>
+        public int GetShaderPropertyId()
+        {
+            if (ShaderPropertyID == -1)
+            {
+                ShaderPropertyID = Shader.PropertyToID(ShaderPropertyName);
+            }
+
+            return ShaderPropertyID;
+        }
+
+        // Properties below are outdated. They remain only for backward compatibility and migration purposes
+        [SerializeField]
+        private int PropId = -1; // i.e OptionIndex
+
+        [SerializeField]
+        private List<ShaderProperties> ShaderOptions = new List<ShaderProperties>();
+
+        [SerializeField]
+        private List<string> ShaderOptionNames = new List<string>();
+
+        [SerializeField]
+        private string ShaderName = "";
+
+        public void MigrateShaderData()
```

Comment:
Is this how you're planning to enable migration? Are we going to expose this somehow to scripts so somebody can just run `updateInteractable.sh` or `updateInteractable.py` to fix up their assets?

Reply:
yes, the shader info just needs to be migrated to the new properties. Shader information was hidden to developers and weirdly auto-generated by the theme inspector. 

I added the menu item: MRTK > Migration > Update Themes to auto execute the update for all assets. That seemed better than having a developer find and figure out how to run a certain script for their project

Also again, if one opens any theme in inspector, it will auto-update the theme.